docs/spec/slashing/future-improvements.md

```diff
@@ -1,4 +1,3 @@
 ## State Cleanup
 
-Once no evidence for a given slashing period can possibly be valid (the end time plus the unbonding period is less than the current time),
-old slashing periods should be cleaned up. This will be implemented post-launch.
+Once a validator is completely unbonded, remove the `ValidatorSigningInfo` from state.
```

Comment:
What prevents a validator from rejoining with their old key and operator address once fully unbonded?

Reply:
I would prevent the same consensus key & operator key from being reused, mostly just to prevent UX confusion.

I don't think the storage is an issue, tombstoning will happen rarely (one would hope).